door-open-service/node-backend/src/StatsDatabase.js

```diff
@@ -45,41 +49,52 @@ DoorOpens.sync();
 class StatsDatabase {
 
   static registerDoorOpen(user) {
-    DoorOpens.create(user);
+    DoorOpens.create({ user: user.name, email: user.email, type: "door" });
   }
 
-  static registerGarageOpen(/*user*/) {
-    // TODO: register garage openings
+  static registerGarageOpen(user) {
+    DoorOpens.create({ user: user.name, email: user.email, type: "garage" });
   }
 
   static getStats() {
-    const queries = {
-      count_per_dayQ: sequelize.query(
-        "SELECT case cast(strftime('%w', timestamp) as integer) " +
-        "when 0 then 'Sun' " +
-        "when 1 then 'Mon' " +
-        "when 2 then 'Tue' " +
-        "when 3 then 'Wed' " +
-        "when 4 then 'Thu' " +
-        "when 5 then 'Fri' " +
-        "else 'Sat' end as dayofweek, COUNT(timestamp) as count FROM door_opens GROUP BY dayofweek", {
-          type: sequelize.QueryTypes.SELECT
-        }),
-      timestampQ: DoorOpens.findOne({
-        order: "timestamp ASC",
-        limit: 1,
-        attributes: ["timestamp"]
-      })
+    const queries = type => {
+      return {
+        count_per_dayQ: sequelize.query(
+          "SELECT case cast(strftime('%w', timestamp) as integer) " +
+            "when 0 then 'Sun' " +
+            "when 1 then 'Mon' " +
+            "when 2 then 'Tue' " +
+            "when 3 then 'Wed' " +
+            "when 4 then 'Thu' " +
+            "when 5 then 'Fri' " +
+            "else 'Sat' end as dayofweek, COUNT(timestamp) as count FROM door_opens " +
+            `WHERE type = '${type}' GROUP BY dayofweek`, {
+              type: sequelize.QueryTypes.SELECT
+            }),
+        timestampQ: DoorOpens.findOne({
+          order: "timestamp ASC",
+          where: { type: type },
+          limit: 1,
+          attributes: ["timestamp"]
+        })
+      };
     };
 
-    return Promise.props(queries)
-      .then(data => {
+    const types = ["door", "garage"];
+
+    return Promise.all(_.map(types, t => Promise.props(queries(t)))).then(data => {
+      return _.map(data, val => {
         const counts0 = _.object(DAYS_OF_WEEK, _.map(DAYS_OF_WEEK, () => 0));
-        const countsDay = _.chain(data.count_per_dayQ).indexBy("dayofweek").mapObject(day => day.count).value();
+        const countsDay = _.chain(val.count_per_dayQ).indexBy("dayofweek").mapObject(day => day.count).value();
         return _.extendOwn(counts0, countsDay, {
-          since: data.timestampQ ? new Date(data.timestampQ.timestamp).toISOString() : null
+          since: val.timestampQ ? new Date(val.timestampQ.timestamp).toISOString() : null
         });
       });
+    }).then(res => {
+      let stats = {};
+      _.each(types, (el, idx) => { stats[el] = res[idx]; });
```

Comment:
I believe underscore's [_.object](http://underscorejs.org/#object) does exactly this, doesn't it?